### PR TITLE
[RU] Added 'with' section translation for the 'basics/control-structures' lesson.

### PR DESCRIPTION
**Pull request by nscyclone**

### CODE REVIEW

File: ru/lessons/basics/control-structures.md [hunk 1]
```diff
@@ -127,3 +128,53 @@ iex> cond do
 ...> end
 "Catch all"
 ```
+
+## `with`
+
+Специальная форма `with` может пригодиться при использовании вложенных операторов `case` или в ситуациях, которые не могут быть просто слиты воедино. `with` состоит из ключевого слова, генераторов и, в конце, выражения.
```
Comment (nscyclone) [inline ru/lessons/basics/control-structures.md]: Actually I assume that there should be a better translation for `situations that cannot cleanly be piped together` → `в ситуациях, которые не могут быть просто слиты воедино` but I unfortunately can't figure it out on my own yet.

**Comment by doomspork** [comment]
You're on top of things @nscyclone!  Thank you

File: ru/lessons/basics/control-structures.md (hunk 1 above)
Comment (brain-geek) [inline ru/lessons/basics/control-structures.md]: As I understand,  'pipe' means pipe operator usage.

What do you think about such option?
"В ситуациях, когда сложно использовать оператор потока, либо когда нужен вложенный вызов case ...."

Comment (nscyclone) [inline ru/lessons/basics/control-structures.md]: Your suggestion is really good in my opinion, @brain-geek . Thanks a lot!
I've just updated it and rebased the branch a moment ago.

File: ru/lessons/basics/control-structures.md [hunk 2]
```diff
@@ -127,3 +128,53 @@ iex> cond do
 ...> end
 "Catch all"
 ```
+
+## `with`
+
+Специальная форма `with` может пригодиться в ситуациях, когда сложно использовать оператор потока, либо когда нужен вложенный вызов `case`. `with` состоит из ключевого слова, генераторов и, в конце, выражения.
```
Comment (brain-geek) [inline ru/lessons/basics/control-structures.md]: в конце, выражения. -> и выражения в конце ?

File: ru/lessons/basics/control-structures.md [hunk 3]
```diff
@@ -127,3 +128,53 @@ iex> cond do
...
+
+Мы ещё обсудим генераторы в уроке о списковых включениях, но сейчас нам достаточно знать лишь то, что они используют сопоставление с образцом для сравнения правой части `<-` с левой.
```
Comment (brain-geek) [inline ru/lessons/basics/control-structures.md]: I think that adding "list comprehension" term in brackets near the first term appearance may be good practice. But it's up to you.

File: ru/lessons/basics/control-structures.md [hunk 4]
```diff
@@ -127,3 +128,53 @@ iex> cond do
...
+```elixir
+iex> user = %{first: "doomspork"}
+%{first: "doomspork"}
+iex> with {:ok, first} <- Map.fetch(user, :first),
+...>      {:ok, last} <- Map.fetch(user, :last),
+...>      do: last <> ", " <> first
+:error
+```
+
+Давайте взглянем на пример побольше без использования `with`, а затем узнаем, как мы могли бы его отрефакторить:
```
Comment (brain-geek) [inline ru/lessons/basics/control-structures.md]: "отрефакторить" sounds like a slang word. Can we use "улучшить" here without losing any information?

Comment (brain-geek) [inline ru/lessons/basics/control-structures.md]: And "бы" sounds a bit obsolete here, if you ask me.

**Comment by nscyclone** [comment]
Thanks for the review, @brain-geek ! I've just applied the fixes that you mentioned before.
Also, if I am not mistaken, earlier you suggested to remove `бы` from `как мы могли бы его отрефакторить`. By now that note magically disappeared so I am wondering whether should we remove the particle or not.

**Comment by brain-geek** [comment]
@nscyclone I've wrote that down, and some time after reconsidered this and removed the comment.

**Comment by brain-geek** [comment]
@nscyclone Thanks for the contribution!

**Pull request merged**